### ISSUE TITLE
Service discovery endpoint doesn't get updated for peer when BIGIPs are in HA 

### ISSUE THREAD

**Issue by vishwajeetgslab** [F5Networks/f5-appsvcs-extension#238]
### Environment
 * Application Services Version:3.17.1
 * BIG-IP Version:15.1.0

### Summary
Service discovery endpoint doesn't get updated for peer when BIGIPs are in HA 
When posted AS3 declaration including Service Discovery config to one BigIP, the same doesnt get updated for its peer in HA.
As a result, when we perform  GET operation on Service discovery endpoint of the peer, we dont get any list of nodes. 
### Steps To Reproduce
Steps :
1. Added 2 ( Physical ) BigIPs in HA Pair ( Say - BigIP21 and BigIP20 )
2. Ensured that AS3 post - excluding Service discovery configurations - worked perfectly. Creation of Partition and App on one BigIP, reflected properly on its peer.
3. Posted AS3 with Service discovery added on BigIP21. Observed that the service discovery endpoint was updated only for BigIP21 and not for BigIP20

POST used : 
Post for Service Discovery config :

 {"class": "ADC", "schemaVersion": "3.17.0", "BasicPart": {"BasicApp": {"class": "Application", "template": "tcp", "serviceMain": {"class": "Service_TCP", "virtualAddresses": ["8.1.8.1"], "virtualPort": 8181, "pool": "svc_pool"}, "svc_pool": {"class": "Pool", "members": [{"servicePort": 8080, "serverAddresses": []}, {"addressDiscovery": "event", "servicePort": 8080}]}, "constants": {"class": "Constants", "serviceCenterEPG": {"svc_pool": {"tenant": "Demo", "application": "AppProfile", "epg": "EPG-Internal9"}}, "appsvcsFormData": {"virtual_address": "8.1.8.1", "virtual_port::integer": 8181, "server_port::integer": 8080, "server_addresses::array": [], "template_name": "simple_tcp"}}}, "class": "Tenant"}, "id": "6e6a0063-054f-4d03-a487-7378bd43d645"} (Thread-Id #126216581400320)
 
Observations :

A) Service discovery endpoint was updated only for BigIP21 and not for BigIP20.

Service discovery endpoint for BigIP21 :
URL : https://BigIP21/mgmt/shared/service-discovery/task/
Observed response : {"code":200,"items":[{"id":"~BasicPart1~AdNaau1iiTaPVgZlU~2VnL7qJV2hKXWEmS562YKyzLY3D","updateInterval":0,"provider":"static","providerOptions":{"type":"static","nodes":[]},"resources":[{"type":"pool","path":"/BasicPart1/BasicApp1/Pool1","options":{"servicePort":8080,"connectionLimit":0,"rateLimit":"disabled","dynamicRatio":1,"ratio":1,"priorityGroup":0,"monitor":"default"},"requiresNodes":true}],"kind":"shared:service-discovery:taskstate","lastRun":"2020-03-12T09:42:15.034Z"},{"id":"~BasicPart1~BasicApp1~Pool1","updateInterval":0,"provider":"event","providerOptions":{"type":"event","nodeList":[{"ip":"5.9.5.11","id":"/BasicPart1/5.9.5.11"},{"ip":"5.9.5.10","id":"/BasicPart1/5.9.5.10"}]},"resources":[{"type":"pool","path":"/BasicPart1/BasicApp1/Pool1","options":{"servicePort":8080,"connectionLimit":0,"rateLimit":"disabled","dynamicRatio":1,"ratio":1,"priorityGroup":0,"monitor":"default"},"requiresNodes":true}],"kind":"shared:service-discovery:taskstate","lastRun":"2020-03-12T09:42:20.651Z","nodePrefix":"/BasicPart1/"}]}

Service discovery endpoint for BigIP20
URL : https://BigIP20/mgmt/shared/service-discovery/task/
Observed response : 
{"code":200,"items":[]}

B) Logs collected from BigIPs for step3 mentioned above:
Logs from BigIPs for step 3 above :
BigIP21 :

restnoded.log for this task :
Thu, 12 Mar 2020 09:42:10 GMT - finest: socket 279 opened
Thu, 12 Mar 2020 09:42:12 GMT - finest: socket 280 opened
Thu, 12 Mar 2020 09:42:12 GMT - fine: [f5-service-discovery: restWorkerTask.js] Got service discovery task request
Thu, 12 Mar 2020 09:42:13 GMT - info: [f5-service-discovery: Task.js] Task ~BasicPart1~AdNaau1iiTaPVgZlU~2VnL7qJV2hKXWEmS562YKyzLY3D failed: The var "nodes" must be instantiated and be an array for comparison
Thu, 12 Mar 2020 09:42:13 GMT - fine: [f5-service-discovery: Task.js] Error: The var "nodes" must be instantiated and be an array for comparison
    at Object.checkNodesExist (/var/config/rest/iapps/f5-service-discovery/nodejs/TmshUtil.js:198:31)
    at StaticSource.createNodes (/var/config/rest/iapps/f5-service-discovery/nodejs/StaticSource.js:21:25)
    at promise.then (/var/config/rest/iapps/f5-service-discovery/nodejs/Task.js:109:70)
    at tryCatcher (/usr/share/rest/node/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromiseCtx (/usr/share/rest/node/node_modules/bluebird/js/release/promise.js:606:10)
    at Async._drainQueue (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:138:12)
    at Async._drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/share/rest/node/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
Thu, 12 Mar 2020 09:42:13 GMT - fine: [f5-service-discovery: restWorkerTask.js] service discovery task succeeded
Thu, 12 Mar 2020 09:42:13 GMT - fine: [f5-service-discovery: restWorkerTask.js] Got service discovery task request
Thu, 12 Mar 2020 09:42:15 GMT - fine: [f5-service-discovery: restWorkerTask.js] service discovery task succeeded
Thu, 12 Mar 2020 09:42:20 GMT - finest: socket 280 closed
Thu, 12 Mar 2020 09:42:27 GMT - finest: socket 279 closed

restjavad for this task :
[I][2201][12 Mar 2020 09:42:10 UTC][8100/shared/identified-devices/config/device-info DeviceInfoWorker] starting GET_DEVICE_INFO
[I][2202][12 Mar 2020 09:42:11 UTC][8100/shared/identified-devices/config/device-info DeviceInfoWorker] starting GET_DEVICE_INFO
[I][2204][12 Mar 2020 09:42:12 UTC][8100/shared/identified-devices/config/device-info DeviceInfoWorker] starting GET_DEVICE_INFO
[WARNING][2222][12 Mar 2020 09:42:16 UTC][8100/tm/shared/bigip-failover-state BigipFailoverStateWorker] Failed to get device-groups from icrd: com.f5.rest.common.RestRequestSender$HttpException: java.net.ProtocolException: status:503, body:{"code":503,"message":"There is an active asynchronous task executing.","errorStack":[],"apiError":32964609}
        at com.f5.rest.common.ApacheAsyncRequestSender$3.run(ApacheAsyncRequestSender.java:528)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:292)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ProtocolException: status:503, body:{"code":503,"message":"There is an active asynchronous task executing.","errorStack":[],"apiError":32964609}
        at com.f5.rest.common.ApacheAsyncRequestSender.handleRequestCompleted(ApacheAsyncRequestSender.java:157)
        at com.f5.rest.common.ApacheAsyncRequestSender$1.completed(ApacheAsyncRequestSender.java:115)
        at com.f5.rest.common.ApacheAsyncRequestSender$1.completed(ApacheAsyncRequestSender.java:111)
        at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:123)
        at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:181)
        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:439)
        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:329)
        at org.apache.http.impl.nio.client.InternalRequestExecutor.inputReady(InternalRequestExecutor.java:83)
        at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
        at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
        at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
        at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
        ... 1 more

[I][2224][12 Mar 2020 09:42:16 UTC][8100/shared/identified-devices/config/device-info DeviceInfoWorker] starting GET_DEVICE_INFO
[I][2226][12 Mar 2020 09:42:21 UTC][8100/shared/identified-devices/config/device-info DeviceInfoWorker] starting GET_DEVICE_INFO

BigIP20
restnoded.log
Thu, 12 Mar 2020 09:40:26 GMT - finest: socket 248 opened
Thu, 12 Mar 2020 09:40:31 GMT - finest: socket 248 closed

### Expected Behavior
The service discovery endpoint should be updated for both the BigIPs when in HA.

### Actual Behavior
Service discovery endpoint doesn't get updated for peer when BIGIPs are in HA.
As a result, when we perform  GET operation on Service discovery endpoint of the peer, we dont get any list of nodes.

**Comment by payalsin** [comment]
Found during testing integration with F5 ACI ServiceCenter

**Comment by kedargslab** [comment]
Just and update to the observations above :
After step 3 above, if we try to POST any New AS3 Partition/App along with Service Discovery config to BigIP20 ( peer) it doesnt work and operation errors out and below error is observed in restnoded.log :

Fri, 13 Mar 2020 06:56:43 GMT - finest: socket 713 opened
Fri, 13 Mar 2020 06:56:44 GMT - warning: [appsvcs] {"message":"unable to digest declaration. Error: GET http://admin:XXXXXX@localhost:8100/mgmt/tm/ltm/node?$select=fullPath,partition,address,ephemeral,metadata,fqdn/tmName&options=recursive query target BIG-IP current ltm node list response=400 body={\"code\":400,\"message\":\"can't get object count from mcpd\",\"errorStack\":[],\"apiError\":26214401}","level":"warning"}
Fri, 13 Mar 2020 06:56:49 GMT - finest: socket 713 closed

**Comment by dstokesf5** [comment]
This appears to be a duplicate of #152. The nodes will not persist without persisting the configuration to disk.
